src/BlockTypes/ProductGalleryLargeImage.php

```diff
@@ -125,7 +125,7 @@ private function get_main_images_html( $context, $product_id ) {
 		);
 
 		if ( $context['fullScreenOnClick'] ) {
-			$attributes['class'] .= ' wc-block-woocommerce-product-gallery-large-image__image--full-screen-on-click';
+			$attributes['class'] .= ' wc-block-woocommerce-product-gallery-large-image__image--full-screen-on-click wc-block-product-gallery-dialog-on-click';
```

Comment:
Is there a reason for this change?

Reply:
We should create another action for the View All link. In this way, we can avoid the check on the CSS class. Does it make sense?

Reply:
It does make sense! I'm open to that for sure.

However, semantically it also makes sense to me to have the same "action" (opening the Dialog) be handled by the same action.

Additionally, `tagName === 'IMG'` seems to be a bit too broad and narrowing it down will make it less fragile to potential conflicts. For example we have the ability to stack/add blocks on top of the Large Image. With the `tagName === 'IMG' check, if you add an image, then the Dialog will be triggered also when that "new" image is clicked.

What do you think?

Reply:
> However, semantically it also makes sense to me to have the same "action" (opening the Dialog) be handled by the same action.

Not really. You can image them as different event handlers for different events. One is the click in the large image and the other one is the click on the View All thumbnails.

> Additionally, tagName === 'IMG' seems to be a bit too broad and narrowing it down will make it less fragile to potential conflicts. For example we have the ability to stack/add blocks on top of the Large Image. 

True, but I think that it is very uncommon use case. For now, I would keep it as it is. However, I don't have a strong opinion, so feel free to keep the check on the class if you think that it makes sense!

Reply:
@roykho Do you have any opinion on this?